### PR TITLE
Adding Express backend barebone

### PR DESCRIPTION
**Pull request by paulsancer**
Adding Express backend barebone, moved react app to /client directory.

Webpack proxy will redirect requests to "/api..." to the backend server at port `5000`.

Run with:

`$ yarn dev`

### CODE REVIEW

**Comment by paulsancer** [comment]
@Bernabe-Felix @BernabeFelix  please review, want to continue with firebase into a new branch

File: controllers/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+
+const app = express();
+const port = process.env.PORT || 5000;
+
+app.get('/api/greet', (req, res) => {
+  res.send({ express: 'Hello From Express' });
+});
+
+app.listen(port, () => console.log(`Listening on port ${port}`));
```
Comment (BernabeFelix) [inline controllers/index.js]: There is not need for this server

File: index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import app from './app';
+
+const port = process.env.PORT || 5000;
+
+app.listen(port, () => console.log(`Listening on port ${port}`));
```
Comment (BernabeFelix) [inline index.js]: this goes into app.js

File: app.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
```
Comment (BernabeFelix) [inline app.js]: This should be called `server.js` intead of `app.js`

File: controllers/AuthController.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import express from 'express';
+import bodyParser from 'body-parser';
+
+const router = express.Router();
+router.use(bodyParser.urlencoded({ extended: true }));
+router.use(bodyParser.json());
+
+// SignUp
+router.post('/', (req, res) => {
```
Comment (BernabeFelix) [inline controllers/AuthController.js]: We need to embrace DRY, so this routes can be chained
```
router.route('/)
 .get(func)
 .post(func)
```

Comment (paulsancer) [reply controllers/AuthController.js]: nice :O

Comment (paulsancer) [reply controllers/AuthController.js]: done

File: app.js (hunk 3 above)
Comment (paulsancer) [reply app.js]: done

File: index.js (hunk 2 above)
Comment (paulsancer) [reply index.js]: done

File: controllers/index.js (hunk 1 above)
Comment (paulsancer) [reply controllers/index.js]: done

**Pull request merged**